File: go/src/infra/libs/clock/testclock/context.go
// Copyright 2015 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package testclock

import (
	"time"

	"golang.org/x/net/context"
	"infra/libs/clock"
)

// UseTime instantiates a TestClock and returns a Context that is configured to
// use that clock, as well as the instantiated clock.
func UseTime(ctx context.Context, now time.Time) (rctx context.Context, tc TestClock) {
	tc = New(now)
	rctx = clock.Set(ctx, tc)
	return
}

File: appengine/findit/findit_v2/services/context.py
# Copyright 2019 The Chromium Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

from libs.structured_object import StructuredObject


class Context(StructuredObject):
  """This class defines the context within which an analysis is run.

  Currently, its main goal is to scope the analysis/data for a specific project
  so that Findit can be more general to support multiple projects.
  """

  # The Luci project name as defined by Buildbucket, e.g. "chromium".
  luci_project_name = basestring

  # The hostname of the gitiles server that hosts the source code,
  # e.g. chromium.googlesource.com
  gitiles_host = basestring

  # The project name in the gitiles server for the code base, e.g.
  # "chromum/src".
  gitiles_project = basestring

  # The ref in the gitiles project/server for the code, e.g. "ref/heads/master".
  gitiles_ref = basestring

  # The sha of a gitiles commit.
  gitiles_id = basestring
